spyder/api/manager.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Copyright © <2013-2016> <Colin Duquesnoy and others, see pyqode/AUTHORS.rst>
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the Manager API.
+Adapted from https://github.com/pyQode/pyqode.core/blob/master/pyqode/core/api/manager.py
+"""
+import weakref
+
+
+class Manager(object):
+    """
+    A manager manages a specific aspect of a CodeEdit instance:
+        - backend management (start/stop server, request work,...)
+        - modes management
+        - panels management and drawing
+        - file manager
+
+    Managers are typically created internally when you create a CodeEdit.
+    You interact with them later, e.g. when you want to start the backend
+    process or when you want to install/retrieve a mode or a panel.
+
+    ::
+        editor = CodeEdit()
+
+        # use the backend manager to start the backend server
+        editor.backend.start(...)
+        editor.backend.send_request(...)
+
+        # use the panels controller to install a panel
+        editor.panels.install(MyPanel(), MyPanel.Position.Right)
+        my_panel = editor.panels.get(MyPanel)
+
+        # and so on
```

Comment:
Please rewrite this docstring to refer to Spyder functionality instead of PyQode one